### PR TITLE
Time: fix bug introduced with #8999

### LINKED ISSUE #9311: Unable to scroll in continuous mode
* KOReader version: 2022.06, both amd64 deb package and appimage version
* Device: Linux PC; debian 11

#### Steps to reproduce
Open an PDF or ePub file, switch the "View Mode" to continuous in the Bottom Bar. And KOReader will crash in this mode if I start scroll.

##### `crash.log` (if applicable)
I can't find `crash.log` file so I paste the console output here.
```
okaypj@okaypj-4287j12:~/.config$ koreader
---------------------------------------------
                launching...
  _  _____  ____                _
 | |/ / _ \|  _ \ ___  __ _  __| | ___ _ __
 | ' / | | | |_) / _ \/ _` |/ _` |/ _ \ '__|
 | . \ |_| |  _ <  __/ (_| | (_| |  __/ |
 |_|\_\___/|_| \_\___|\__,_|\__,_|\___|_|

 It's a scroll... It's a codex... It's KOReader!

 [*] Current time: 07/09/22-07:29:46
 [*] Version: v2022.06

ffi.load: blitbuffer
ffi.load: SDL2
ffi.load: SDL2
Starting SDL in /usr/lib/koreader/
SDL: no gamecontrollers connected
ffi.load: libs/liblodepng.so
07/09/22-07:29:46 INFO  initializing for device Linux
07/09/22-07:29:46 INFO  framebuffer resolution: {
    ["w"] = 1366,
    ["h"] = 711,
}
ffi.load: libs/libmupdf.so
ffi.load: libs/libwrap-mupdf.so
ffi.load: libs/libfreetype.so.6
ffi.load: libs/libharfbuzz.so.0
ffi.load: libs/libzstd.so.1
ffi.load: libs/libutf8proc.so.2
07/09/22-07:29:46 INFO  Loading plugins from directory: plugins
ffi.load: sqlite3
07/09/22-07:29:47 INFO  Loading plugins from directory: /home/okaypj/.config/koreader/plugins/
07/09/22-07:29:56 INFO  opening file _(my file here)_
CRE WARNING: cached rendering is invalid (stylesheet hash mismatch): doing full rendering
/usr/lib/koreader/luajit: frontend/apps/reader/modules/readerrolling.lua:496: attempt to perform arithmetic on field 'time' (a table value)
stack traceback:
	frontend/apps/reader/modules/readerrolling.lua:496: in function 'handleEvent'
	frontend/ui/widget/container/widgetcontainer.lua:95: in function 'propagateEvent'
	frontend/ui/widget/container/widgetcontainer.lua:113: in function 'handleEvent'
	frontend/ui/uimanager.lua:1136: in function 'broadcastEvent'
	frontend/device/sdl/device.lua:217: in function 'handleSdlEv'
	frontend/device/input.lua:1254: in function 'waitEvent'
	frontend/ui/uimanager.lua:1670: in function 'handleInput'
	frontend/ui/uimanager.lua:1726: in function 'run'
	./reader.lua:324: in main chunk
	[C]: at 0x564585e08b90

```

### PR DESCRIPTION
**Pull request by zwim**
This should fix #9311.

Draft as only tested on the emulator by now.

---
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/9314)

### CODE REVIEW

File: frontend/device/input.lua [hunk 1]
```diff
@@ -1208,34 +1214,33 @@ function Input:waitEvent(now, deadline)
...
-                        event.time.sec, event.time.usec))
+                        time.format_time(event.time)))
                 elseif event.type == C.EV_SYN then
                     logger.dbg(string.format(
-                        "input event => type: %d (%s), code: %d (%s), value: %s, time: %d.%06d",
-                        event.type, linux_evdev_type_map[event.type], event.code, linux_evdev_syn_code_map[event.code], event.value,
-                        event.time.sec, event.time.usec))
+                        "input event => type: %d (%s), code: %d (%s), value: %s, time: %s",
+                        event.type, linux_evdev_type_map[event.type], event.code, linux_evdev_syn_code_map[event.code],
+                        event.value,time.format_time(event.time)))
```
Comment (poire-z) [inline frontend/device/input.lua:1224]: Just asking: how worthy/costly is it to do all these `time.format_time()`, here for logger.dbg() that won't show up on devices, and above for all events (are the ev.time always used, or only when needed like for this scrolling thing, in which case isn't it costly ?)

Comment (zwim) [reply frontend/device/input.lua:1224]: One `time.format_time` does a string format (for float) and one multiplication.

In `dbg.lua` one string concatenation is saved.

In `input.lua` two string formats (for int) are saved.

So overall I see almost no cost impact. (In fact the overhead introduced in `UIManager` is saved in `handleTouchEv`.)

About the worthiness of the logged infos @NiLuJe should be asked.

`ev.time` is needed to detect gestures in `gesturedetector.lua`.

Comment (NiLuJe) [reply frontend/device/input.lua:1224]: I'd rather figure out where the non-time one is seeping through, as, yeah, that's a "lot" of overhead when we only ever have a *single* time per event stream (the SYN_REPORT one).

Comment (NiLuJe) [reply frontend/device/input.lua:1224]: (Speaking solely for the common code-path, the debug stuff is entirely harmless as hidden behind an if dbg branch. And I rely on it being that verbose. A lot.).

**Comment by zwim** [comment]
The last commit fixes the bug. But apparently this is only for SDL.

I don't know if the bug also exists on other devices.

**Comment by Frenzie** [comment]
Given that it's only when scrolling I would guess not, given that I wrote this (i.e., there's a chance it's simply outdated? it hasn't been meaningfully updated in three years; not that there was any reason to until your recent changes):

https://github.com/koreader/koreader/blob/7366a031d6fc0e9d81d313ec859783a27bf6f152/frontend/device/sdl/device.lua#L207-L236'

**Comment by Frenzie** [comment]
It's basically just a quick hack that works reasonably well since I didn't want to spend a lot of time implementing completely separate scrolling support when I could get 98% of the way there in 10-20 minutes.

**Comment by zwim** [comment]
Then it should be more stringent to change the
```
time = ev.time
```
to
```
time = time.timeval(ev.time)
```
in `sdl/device.lua` as to mess in `input.lua`.

**Review by Frenzie — APPROVED** [review]
Yes, this is what I thought needed to happen, just wasn't sure if that was the right way to do it. Since as @poire-z said, I don't want to be bothered trying to understand new time yet.

**Review by poire-z — APPROVED** [review]

**Pull request merged**